master/buildbot/process/properties.py

```diff
@@ -743,3 +743,27 @@ def getRenderingFor(self, build):
         return d
 
 registerAdapter(_DictRenderer, dict, IRenderable)
+
+
+class Transform(object):
+
+    """
+    A renderable that combines other renderables' results using an arbitrary function.
+    """
+
+    implements(IRenderable)
+
+    def __init__(self, function, *args, **kwargs):
+        if not callable(function) and not IRenderable.providedBy(function):
+            config.error("function given to Transform neither callable nor renderable")
+
+        self.function = function
+        self.args = args
+        self.kwargs = kwargs
+
+    @defer.inlineCallbacks
+    def getRenderingFor(self, iprops):
+        rfunction = yield iprops.render(self.function)
```

Comment:
No, it's not about the name, I do not quite understand the idea behind `yield iprops.render(self._function)`...